azkaban-common/src/main/java/azkaban/project/JdbcProjectLoader.java

```diff
@@ -382,6 +382,25 @@ private void uploadProjectFile(Connection connection, Project project,
     }
 
     logger.info("Md5 hash created");
+
+    /**
+     * Insert a version row to table project_versions.
+     * If something goes wrong during uploading project files to DB, the latest version number
+     * in project_versions refers to the destructive project_file. Since AZ always let the number+1
```

Comment:
Do you mean " this is desirable to ensure that the next version number will be unique"?